python/fast_mlsirm/estimators/marginal.py

```
@@ -1720,12 +1860,19 @@ def estep(current_params):
     converged = False
     final_delta = np.inf
     stopping_tolerance = float(tol * (1.0 + abs(ll)))
+
+    # Optimized: precompute a one-hot category mask outside the iterative loop to eliminate intermediate allocations
+    y_mask = (y[:, :, None] == scores).astype(np.float64)
+
     for it in range(1, max_iter + 1):
```

Ruff (0.16.1)

[warning] Loop control variable `it` not used within loop body

(B007)